cin_validator/rules/cin2022_23/rule_8750.py

```diff
@@ -0,0 +1,166 @@
+from cin_validator.test_engine import run_rule
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+
+ChildIdentifiers = CINTable.ChildIdentifiers
+LAchildID = ChildIdentifiers.LAchildID
+PersonBirthDate = ChildIdentifiers.PersonBirthDate
+ExpectedPersonBirthDate = ChildIdentifiers.ExpectedPersonBirthDate
+GenderCurrent = ChildIdentifiers.GenderCurrent
+
+# define characteristics of rule
+@rule_definition(
+    # write the rule code here, in place of 8500
+    code=8750,
+    # replace ChildIdentifiers with the value in the module column of the excel sheet corresponding to this rule .
+    module=CINTable.ChildIdentifiers,
+    # replace the message with the corresponding value for this rule, gotten from the excel sheet.
+    message="Gender must equal 0 for an unborn child",
+    # The column names tend to be the words within the < > signs in the github issue description.
+    affected_fields=[GenderCurrent, PersonBirthDate],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # PREPARING DATA
+
+    # Replace ChildIdentifiers with the name of the table you need.
+    df = data_container[ChildIdentifiers]
+    # Before you begin, rename the index so that the initial row positions can be kept intact.
+    df.index.name = "ROW_ID"
+
+    # lOGIC
+    # Implement rule logic as described by the Github issue.
+    # Put the description as a comment above the implementation as shown.
+
+    # If <ExpectedPersonBirthDate> (N00098) is present and <PersonBirthDate> (N00066) is blank then <GenderCurrent> (N00065) must equal “0”
+    condition = (
+        df[PersonBirthDate].isna()
+        & df[ExpectedPersonBirthDate].notna()
+        & (df[GenderCurrent].astype(str) != "0")
+    )
+    # get all the data that fits the failing condition. Reset the index so that ROW_ID now becomes a column of df
+    df_issues = df[condition].reset_index()
+
+    # SUBMIT ERRORS
+    # Generate a unique ID for each instance of an error. In this case,
+    # - If only LAchildID is used as an identifier, multiple instances of the error on a child will be understood as 1 instance.
+    # We don't want that because in reality, a child can have multiple instances of an error.
+    # - If we use the LAchildID-CPPstartDate combination, that artificially cancels out the instances where a start date repeats for the same child.
+    # Another rule checks for that condition. Not this one.
+    # - It is very unlikely that a combination of LAchildID-CPPstartDate-CPPendDate will repeat in the DataFrame.
+    # Hence, it can be used as a unique identifier of the row.
+
+    # Replace CPPstartDate and CPPendDate below with the columns concerned in your rule.
+    link_id = tuple(
+        zip(
+            df_issues[LAchildID],
+            df_issues[GenderCurrent],
+            df_issues[PersonBirthDate],
+        )
+    )
```

Comment:
2/3 
As mentioned in the note, the ERROR_ID doesn't need to contain PersonBirthDate. It can be formed from a combination of 
- LAchildID and ExpectedPersonBirthDate (preferred)
- LAchildID and GenderCurrent
but not the three. The former combination is preferred because it fits the guideline of creating ERROR_ID from the "condition" columns. However, the latter combination is also appropriate in this case.

Reply:
Afraid I still don't quite understand this.  If we assume that _somehow_ an LA has loaded multiple rows for the same child into the one-row-per-child ChildIdentifiers table, you could end up with, for example:

| ChildID | ExpectedDOB | ActualDOB | Gender |
| -------- | -------------- | ------------- | -------- |
| Child 1 | 10/10/2020 | NULL | 0 |
| Child 1 | 10/10/2020 | NULL | 1 |
| Child 1 | 10/10/2020 | NULL | 2 |
| Child 1 | 20/10/2020 | NULL | 2 |

(Probably not an exhaustive list)

That's potentially multiple in-error rows with the same EDD and different gender and multiple rows with different EDD and same gender all which would need to be listed out separately.

From the note that makes me think this would need to zip all three fields (not including ActualDOB) to cover all possibilities for distinct rows?